src/WSSlots.php

```diff
@@ -204,4 +275,48 @@ final public static function getSlotContent( WikiPage $wikiPage, string $slot ):
 
 		return $revisionRecord->getContent( $slot );
 	}
+
+	/**
+	 * Return true if the page should be watched, false otherwise.
+	 *
+	 * @param string $watchlist Valid values: 'watch', 'unwatch', 'preferences', 'nochange'
+	 * @param Title $title The page under consideration
+	 * @param User $user The user get the value for
+	 * @return bool
+	 */
+	protected static function getWatchlistValue(
+		string $watchlist,
+		Title $title,
+		User $user,
+	): bool {
+		$services = MediaWikiServices::getInstance();
+
+		$userWatching = $services->getWatchlistManager()->isWatchedIgnoringRights( $user, $title );
```

Comment:
It does not make sense to directly use `$services->getWatchlistManager()` here, but a bit higher up only call it with `method_exists( MediaWikiServices::class, 'getWatchlistManager' )` around it.